core/rfb.js

```diff
@@ -1623,6 +1627,31 @@ export default class RFB extends EventTargetMixin {
         return this._fail("No supported sub-auth types!");
     }
 
+    _negotiateRA2neAuth() {
+        if (this._rfbRA2State === null) {
+            this._rfbRA2State = new RA2State(this._sock, () => {
+                this._rfbInitState = "SecurityResult";
+                this._rfbRA2State = null;
+                this._initMsg();
+            }, () => this._rfbCredentials, (hasUsername, fingerprint) => {
+                if (hasUsername) {
+                    this.dispatchEvent(new CustomEvent(
+                        "credentialsrequired",
+                        { detail: { types: ["username", "password"], fingerprint: fingerprint } }));
```

Comment:
This API change needs to be documented as well.

And after thinking a bit more about things, I think it is better if we create a new event type for this. It avoids confusing things.